src/IO/WriteBufferFromS3BufferAllocationPolicy.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "config.h"
+
+#if USE_AWS_S3
+
+#include <Storages/StorageS3Settings.h>
+
+namespace DB
+{
+
+struct IBufferAllocationPolicy
```

Comment:
This class is related only to internal stuff inside WriteBufferFromS3, so it seems like it should be declared inside `WriteBufferFromS3`.